polytracker/mapping.py

```diff
@@ -4,24 +4,59 @@
 
 from collections import defaultdict
 from pathlib import Path
-from typing import Dict, List, Set, Tuple, Generator
+from typing import Dict, List, Set, Tuple, Iterator
 from tqdm import tqdm
-from mmap import mmap, PROT_READ
 
 from .plugins import Command
-from .taint_dag import TDFile, TDRangeNode, TDSourceNode, TDUnionNode
+from .taint_dag import TDFile, TDNode, TDRangeNode, TDSourceNode, TDUnionNode
 
 
+LabelType = int
 OffsetType = int
+FileOffsetType = Tuple[Path, OffsetType]
 CavityType = Tuple[OffsetType, OffsetType]
 
 
 class InputOutputMapping:
     def __init__(self, f: TDFile):
         self.tdfile: TDFile = f
 
-    def mapping(self) -> Dict[OffsetType, Set[OffsetType]]:
-        raise NotImplementedError()
+    def dfs_walk(
+        self, label: LabelType, seen: Set[LabelType] = set()
```

Comment:
Isn't the `seen = set()` part affected by https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments ?

Reply:
Thanks! TIL!